legal-api/src/legal_api/services/permissions.py

```diff
@@ -0,0 +1,179 @@
+# Copyright © 2025 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=too-many-lines
+"""This manages all of the authorization service."""
+from enum import Enum
+
+from flask import current_app, g
+
+from legal_api.core.filing import Filing as CoreFiling
+from legal_api.models.authorized_role_permission import AuthorizedRolePermission
+from legal_api.services import authz
+from legal_api.services.cache import cache
+from legal_api.services.filings.validations.dissolution import DissolutionTypes
+
+
+class ListFilingsPermissionsAllowed(str, Enum):
+    """Define an enum for permissions checks."""
+
+    ADDRESS_CHANGE_FILING = 'ADDRESS_CHANGE_FILING'
+    AGM_CHG_LOCATION_FILING = 'AGM_CHG_LOCATION_FILING'
+    AGM_EXTENSION_FILING = 'AGM_EXTENSION_FILING'
+    ALTERATION_FILING = 'ALTERATION_FILING'
+    AMALGAMATION_FILING = 'AMALGAMATION_FILING'
+    ANNUAL_REPORT_FILING = 'ANNUAL_REPORT_FILING'
+    CONSENT_AMALGAMATION_OUT_FILING = 'CONSENT_AMALGAMATION_OUT_FILING'
+    CONSENT_CONTINUATION_OUT_FILING = 'CONSENT_CONTINUATION_OUT_FILING'
+    CONTINUATION_IN_FILING = 'CONTINUATION_IN_FILING'
+    CORRECTION_FILING = 'CORRECTION_FILING'
+    COURT_ORDER_FILING = 'COURT_ORDER_FILING'
+    DISSOLUTION_ADMIN_FILING = 'ADMIN_DISSOLUTION_FILING'
+    DISSOLUTION_DELAY_FILING = 'DELAY_DISSOLUTION_FILING'
+    DISSOLUTION_FIRM_FILING = 'FIRM_DISSOLUTION_FILING'
+    DISSOLUTION_INVOLUNTARY_FILING = 'DISSOLUTION_INVOLUNTARY_FILING'
```

Comment:
Should we include this one, since we don't have a permission for it? (I mean, this could be added later if we add a permission for it.) (